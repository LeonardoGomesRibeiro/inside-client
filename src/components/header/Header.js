import React from 'react';
import PropTypes from 'prop-types';
import { Layout, Icon, Menu, Badge } from 'antd';
import { connect } from 'react-redux';

import { PROFILE } from '../../mock/profile';
import { logoutRequestAction } from '../../features/auth/actions/authActions';

import './Header.scss';

const { Header } = Layout;
const { SubMenu } = Menu;
const { user: UserMock } = PROFILE;

class MyHeader extends React.Component {
  constructor(props) {
    super(props);

    this.onLogoutClick = this.onLogoutClick.bind(this);
  }

  onLogoutClick() {
    this.props.logoutRequestAction();
  }

  render() {
    const { collapsed, toggleMenu } = this.props;
    return (
      <Header className="header" style={{ background: '#fff', padding: 0 }}>
        <div>
          <Icon
            className="trigger"
            type={collapsed ? 'menu-unfold' : 'menu-fold'}
            onClick={toggleMenu}
          />
        </div>

        <div className={'rightWrapper'}>
          <div className={'button'}>
            <Badge count={0}>
              <Icon type="bell" />
            </Badge>
          </div>

          <div className={'button'}>
            <Badge count={1}>
              <Icon type="message" />
            </Badge>
          </div>

          <Menu mode="horizontal">
            <SubMenu
              title={
                <span>
                  <Icon type="user" />
                  {UserMock.nickname}
                </span>
              }
            >
              <Menu.Item key="edit_rofile">
                <Icon type="edit" /> Editar Perfil
              </Menu.Item>

              <Menu.Item key="settings">
                <Icon type="setting" /> Configurações
              </Menu.Item>

              <Menu.Item key="logout">
                <div onClick={this.onLogoutClick}>
                  <Icon type="logout" />
                  <span>Logout</span>
                </div>
              </Menu.Item>

              <Menu.Item key="about">
                <Icon type="info-circle-o" /> Sobre
              </Menu.Item>
            </SubMenu>
          </Menu>
        </div>
      </Header>
    );
  }
}

MyHeader.propTypes = {
  collapsed: PropTypes.bool.isRequired,
  toggleMenu: PropTypes.func.isRequired,
  logoutRequestAction: PropTypes.func.isRequired,
};

export default connect(null, { logoutRequestAction })(MyHeader);
