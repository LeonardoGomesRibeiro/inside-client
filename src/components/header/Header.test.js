import React from 'react';
import { shallow } from 'enzyme';
import { HeaderWithoutRedux } from './Header';

describe('Header', () => {
  const mockToggleMenu = jest.fn();
  const props = { collapsed: false, toggleMenu: mockToggleMenu };
  const header = shallow(<HeaderWithoutRedux {...props} />);

  it('renders properly', () => {
    expect(header).toMatchSnapshot();
  });
});
