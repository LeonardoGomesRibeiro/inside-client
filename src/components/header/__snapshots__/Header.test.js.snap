// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyHeader
    collapsed={false}
    toggleMenu={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Icon
        className="trigger"
        onClick={[Function]}
        type="menu-fold"
/>,
      "style": Object {
        "background": "#fff",
        "padding": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "className": "trigger",
        "onClick": [Function],
        "type": "menu-fold",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          className="trigger"
          onClick={[Function]}
          type="menu-fold"
/>,
        "style": Object {
          "background": "#fff",
          "padding": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "trigger",
          "onClick": [Function],
          "type": "menu-fold",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
